src/core/fee-calculation/balance-fetcher.ts

```
@@ -121,12 +121,21 @@ export class BalanceFetcher {
 
     // Count only reward distributors for accurate progress tracking
     const rewardDistributors = distributorEntries.filter(
-      ([, info]) => info && info.is_reward_distributor,
+      ([, infoArray]) => infoArray && infoArray[0]?.is_reward_distributor,
     );
     let rewardDistributorIndex = 0;
 
-    for (const [address, distributorInfo] of distributorEntries) {
-      if (!distributorInfo) continue;
+    for (const [address, distributorInfoArray] of distributorEntries) {
+      if (!distributorInfoArray || distributorInfoArray.length === 0) continue;
+
+      const distributorInfo = distributorInfoArray[0];
+
+      if (!distributorInfo) {
+        console.log(
```

Again, should throw an error

---

Fixed - Changed to throw an error here as well, maintaining consistency with the recipient-received-scanner change.